mkdocs/search.py

```diff
@@ -26,9 +26,9 @@ def _find_toc_by_id(self, toc, id_):
         for toc_item in toc:
             if toc_item.url[1:] == id_:
                 return toc_item
-            for toc_sub_item in toc_item.children:
-                if toc_sub_item.url[1:] == id_:
-                    return toc_sub_item
+            toc_item_r = self._find_toc_by_id(toc_item.children, id_)
+            if toc_item_r is not None:
+                return toc_item_r
```

Comment:
That wouldn't work because recursion breaks as soon as it reaches the first empty list of children. For example:

```
() recursing [u'#_1', u'#_2'] for dnsmasq
(_1) recursing [u'#capwap-split-mac', u'#local-switching-vs-overlay-tunnels', u'#vlans-vlan-profiles-overlayvlanwla-tunnels'] for dnsmasq
(capwap-split-mac) recursing [] for dnsmasq
(_1) recursion returned <type 'NoneType'>
() recursion returned <type 'NoneType'>
```

So `_find_toc_by_id` returns `None`.